### PR TITLE
Use in-repo addon's "root" property for includes location

### PR DESCRIPTION
**Pull request by mdeanjones**
The goal is to instrument in-repo addons that don't reside directly within `/lib` by using the addon's root property, if available. 

Given how long the current implementation has been around I think I might be missing some important context that makes this less viable than a quick glance would suggest.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -169,7 +169,11 @@ module.exports = {
    */
   _getIncludesForInRepoAddonDirectories() {
     return this._findInRepoAddons().reduce((acc, addon) => {
-      let addonDir = path.join(this.project.root, 'lib', addon.name);
+      let addonDir =
+        typeof addon.root === 'string'
+          ? addon.root
```
Comment (rwjblue) [inline index.js:174]: Hmm, why would `addon.root` not be present? I would think it always would...

File: index.js [hunk 2]
```diff
@@ -169,7 +169,11 @@ module.exports = {
...
+          : path.join(this.project.root, 'lib', addon.name);
```
Comment (rwjblue) [inline index.js:175]: I agree, this seems super odd / not good. Assuming that an addon is in `lib/some-name` is definitely broken.

**Review by rwjblue — COMMENTED** [review]
Overall, this definitely seems like the right direction! I'd like to land #289 first (to avoid that larger PR having too much rebase hell), but we should definitely get this in...

**Comment by rwjblue** [comment]
OK, landed #289 so this is ready for a rebase. Note the testing system has changed a bit (not using `ember-cli-addon-tests` any more, for example), so the test might need to be tweaked a little more but overall I think the newer system is much easier to understand/work with...

**Comment by mdeanjones** [comment]
Done, and might I add a very pleasant evolution to the testing story. The hesitance of touching anything `/lib` related was an overabundance of caution. It appears as though `addon.root` has been there since time immemorial but I didn't want to bork any edge-cases just in case.

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you @mdeanjones!

File: test-packages/my-app/custom-dir-in-repo-addon.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs-extra');
+const fixturify = require('fixturify');
+
+class InRepoAddon {
```
Comment (Turbo87) [inline test-packages/my-app/custom-dir-in-repo-addon.js:7]: we don't need this file anymore, right?

**Comment by apellerano-pw** [comment]
I have a question on how to exclude an in-repo-addon introduced into the coverage report by this PR.

We have a folder structure like
```
- client
  - our-app
  - design-system
```

And in our-app's package.json we specify the in-repo-addon with `../design-system` in the `ember-addon.paths` array.

How can I exclude this in-repo-addon from the coverage report? I've tried even `**/design-system/**` which i thought would be the glob catch-all but that didn't seem to do it.
